### ISSUE TITLE
Change how we ignore messages

### ISSUE THREAD

**Issue by McInkay** [thelounge/thelounge#2581]
As mentioned in #2573, we probably want to change how we ignore messages.

Currently, they are just dropped on the server. What would be nice is if they are marked as "ignored" and then the client ignores them (either in JS or css, I'm sure we can bikeshed about that).

**Comment by xPaw** [comment]
This approach would introduce many new problems that would need to be solved.

Particularly, if you receive a direct message, Lounge creates a new channel window, but you wouldn't want to do that with ignored users.

**Comment by McInkay** [comment]
Yeah, it's definitely a more involved way of doing it, which will bring up more issues, but do you agree that in the end it's a more elegant situation as it means we can allow people to turn off the ignore and get the ignored messages back.

**Comment by astorije** [comment]
I totally agree with the concept. The current solution we have is simple and works but has limitations and was not meant to be long-term.
Just like most things, it's more complex but offers better support, and I can see how this will also be beneficial for future improvements to the feature.

You are right @xPaw that there are several cases / places to cover. Do we want to use this issue (@McInkay's original description) to try and list them? I think it would be helpful to whoever wants to pick it up.

**Comment by McInkay** [comment]
I've added some of the things I can think of to the description. Feel free to list others I've missed.
